ipatests/test_integration/test_replica_promotion.py

```diff
@@ -702,7 +702,7 @@ def test_replica_install_for_custodia(self):
         tasks.install_replica(master, replica1, setup_ca=False)
         replica1.run_command(['ipactl', 'status'])
         replica1.run_command(['systemctl', 'stop', 'ipa-custodia'])
-        replica1.run_command(['ipactl', 'status'])
+        replica1.run_command(['ipactl', 'status'], raiseonerr=False)
 
```

Comment:
I realize you're just fixing the failure but I'm having a hard time understanding the purpose of this test.

Ostensibly it is testing that keys can be retrieved according to https://github.com/freeipa/freeipa/commit/dcaa62f6a4e0a57de9d0affda584a27539bc4a36

But since this server doesn't have a CA the replica installer will pick the first server as the ca_host and completely avoid the second one. I guess this is reinforced by the fact that custodia is stopped and the install is still successful, but that is a pretty subtle difference.

Either way the ipactl status on lines 703 and 705 aren't really needed and don't really do anything other than check that 1) after an IPA installation all services are running and 2) ipa-custodia is stoppable.

Reply:
IIRC during replica installation, the replica could be talking to custodia on 2 different masters if the selected master did not host a CA instance. The test wanted to make sure that all communications went to the same custodia instance.
I agree with you that the ipactl status lines aren't needed, they are just helpful for debugging. I can remove them if you prefer.

Reply:
Yeah, I can see it being helpful during troubleshooting. In that light it's fine with your change.